setup.cfg

```diff
@@ -94,6 +94,6 @@ development =
     pytzdata
 pandas =
     pandas>=2.1.2,<3.0.0
-    pyarrow<19.0.0
+    pyarrow>=10.0.1,<19.0.0
```

Comment:
@sfc-gh-mkubik  The current supported PyArrow release is version 20. Why the restriction to <19 here? That forces an outdated and unsupported PyArrow release. Pandas has no such restriction, so wondering why this is here?

I actually just raised #2349 about this.